### PR TITLE
Add helpful links in warning and error logs

### PR DESCRIPTION
**Pull request by JasonFreeberg**
Building off William's recent PR to add "WARNING" and "ERROR" labels: https://github.com/microsoft/Oryx/pull/1099

This PR adds helpful troubleshooting links in the log messages to expedite a user's troubleshooting experience

### CODE REVIEW

File: src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl [hunk 1]
```diff
@@ -92,6 +92,7 @@ fi
         pythonBuildExitCode=${PIPESTATUS[0]}
         if [[ $pythonBuildExitCode != 0 ]]
         then
+            LogWarning "Failed to install poetry with exist status ${pythonBuildExitCode}. More information: https://aka.ms/troubleshoot-python"
```
Comment (gpcastro) [inline src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl:95]: "exit code" for consistency?

**Review by gpcastro — APPROVED** [review]

File: src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl (hunk 1 above)
Comment (JasonFreeberg) [reply src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl:95]: Done!

**Comment by qianz2** [comment]
@JasonFreeberg Could you rebase this branch on main to make this PR up-to-date. After that, we'll trigger the validation builds for it. Thanks

**Comment by JasonFreeberg** [comment]
@qianz2 -- Done!

**Pull request merged**